numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1850,18 +1853,35 @@ gentype_reduce(PyObject *self, PyObject *NPY_UNUSED(args))
     if (ret == NULL) {
         return NULL;
     }
+
 #if defined(NPY_PY3K)
     if (PyArray_IsScalar(self, Unicode)) {
         /* Unicode on Python 3 does not expose the buffer interface */
         buffer = PyUnicode_AS_DATA(self);
         buflen = PyUnicode_GET_DATA_SIZE(self);
     }
-    else
-#endif
+    else if (PyObject_GetBuffer(self, &view, PyBUF_SIMPLE) >= 0) {
+        buffer = view.buf;
+        buflen = view.len;
+        /*
+         * XXX: PyObject_AsWriteBuffer does also this, but it is unsafe, as
```

Comment:
Do you mean `AsReadBuffer`? Are you saying that `PyBuffer_Release` is unsafe too, or just that `AsReadBuffer` is?

Reply:
That should have been changed. Both release the buffer.